File: engine/engine-model/src/main/scala/tools/packet/guild/GuildMarkChanged.scala
package tools.packet.guild

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class GuildMarkChanged(private var _characterId: Int, private var _logoBackground: Int,
                       private var _logoBackgroundColor: Int, private var _logo: Int, private var _logoColor: Int) extends PacketInput {
  def characterId: Int = _characterId

  def logoBackground: Int = _logoBackground

  def logoBackgroundColor: Int = _logoBackgroundColor

  def logo: Int = _logo

  def logoColor: Int = _logoColor

  override def opcode(): SendOpcode = SendOpcode.GUILD_MARK_CHANGED
}

File: engine/engine-model/src/main/java/tools/packet/guild/GuildMarkChanged.java
package tools.packet.guild;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record GuildMarkChanged(Integer characterId, Integer logoBackground, Integer logoBackgroundColor, Integer logo,
                               Integer logoColor) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.GUILD_MARK_CHANGED;
   }
}